app/models/self_employed_account.rb

```diff
@@ -1,5 +1,5 @@
 class SelfEmployedAccount < Account
-
   field :hourly_rate, type: BigDecimal, default: 0.0
 
+  validates_format_of :hourly_rate, with: /\A\d+(?:\.\d{0,2})?\z/
```

Comment:
Prefer the new style validations `validates :column, format: value` over `validates_format_of`.